### PR TITLE
[Mesos] expand coarse-grained mode docs

### PR DESCRIPTION
**Pull request by mgummelt**
## What changes were proposed in this pull request?

docs
## How was this patch tested?

viewed the docs in github

### CODE REVIEW

**Comment by mgummelt** [comment]
I've added clarity to the coarse-grained mode documentation, as well as fixed some inconsistencies it had with the new Mesos scheduler

**Comment by holdenk** [comment]
Even for small doc changes it can be useful (to speed up review) to verify that the docs build as expected and mention that under tests.

**Comment by mgummelt** [comment]
I added that I've verified the docs render correctly in github

File: docs/running-on-mesos.md [hunk 1]
```diff
@@ -180,30 +180,47 @@ Note that jars or python files that are passed to spark-submit should be URIs re
...
+multi-tenant cluster, including one which runs multiple concurrent
+Spark applications.
+
+The scheduler will start executors round-robin on the offers Mesos
+gives it, but there are no spread guarantees, as Mesos does not
+provide such guarantees on the offer stream.
+
+The benefit of coarse-grained mode is much lower startup overhead, but
+at the cost of reserving Mesos resources for the complete duration of
+the application.
```
Comment (rxin) [inline docs/running-on-mesos.md]: say a word about dynamic allocation?\

**Comment by rxin** [comment]
LGTM except the minor comment.

File: docs/running-on-mesos.md [hunk 2]
```diff
@@ -180,30 +180,47 @@ Note that jars or python files that are passed to spark-submit should be URIs re
...
-using `conf.set("spark.cores.max", "10")` (for example).
-
-In "fine-grained" mode, each Spark task runs as a separate Mesos task. This allows
-multiple instances of Spark (and other frameworks) to share machines at a very fine granularity,
-where each application gets more or fewer machines as it ramps up and down, but it comes with an
-additional overhead in launching each task. This mode may be inappropriate for low-latency
-requirements like interactive queries or serving web requests.
+Spark can run over Mesos in two modes: "coarse-grained" (default) and
+"fine-grained".
+
```
Comment (sascala) [inline docs/running-on-mesos.md]: ## Coarse-Grained Mode

File: docs/running-on-mesos.md [hunk 3]
```diff
@@ -180,30 +180,47 @@ Note that jars or python files that are passed to spark-submit should be URIs re
...
+Spark applications.
+
+The scheduler will start executors round-robin on the offers Mesos
+gives it, but there are no spread guarantees, as Mesos does not
+provide such guarantees on the offer stream.
+
+The benefit of coarse-grained mode is much lower startup overhead, but
+at the cost of reserving Mesos resources for the complete duration of
+the application.
+
```
Comment (sascala) [inline docs/running-on-mesos.md]: ## Fine-Grained Mode

File: docs/running-on-mesos.md [hunk 4]
```diff
@@ -212,7 +229,9 @@ To run in fine-grained mode, set the `spark.mesos.coarse` property to false in y
 conf.set("spark.mesos.coarse", "false")
 {% endhighlight %}
 
-You may also make use of `spark.mesos.constraints` to set attribute based constraints on mesos resource offers. By default, all resource offers will be accepted.
+You may also make use of `spark.mesos.constraints` to set attribute
+based constraints on mesos resource offers. By default, all resource
+offers will be accepted.
```
Comment (sascala) [inline docs/running-on-mesos.md]: **attribute-based** constraints on **Mesos**

**Comment by mgummelt** [comment]
@rxin ready for merge.  Would also like this in the 2.0 branch if possible.

File: docs/running-on-mesos.md [hunk 5]
```diff
@@ -180,30 +180,53 @@ Note that jars or python files that are passed to spark-submit should be URIs re
...
+
+The benefit of coarse-grained mode is much lower startup overhead, but
+at the cost of reserving Mesos resources for the complete duration of
+the application.  To configure your job to dynamically adjust to its
+resource requirements, look into
+[Dynamic Allocation](#dynamic-resource-allocation-with-mesos).
+
+## Fine-Grained
+
+In "fine-grained" mode, each Spark task inside the Spark executor runs
```
Comment (rxin) [inline docs/running-on-mesos.md]: can you add a message saying this is deprecated as of spark 2.0?

Comment (mgummelt) [inline docs/running-on-mesos.md]: I just made a separate PR for that: https://github.com/apache/spark/pull/14078

Does that work?

Comment (rxin) [inline docs/running-on-mesos.md]: yup both should go into 2.0.

**Pull request closed without merging**